storage/storage.go

```diff
@@ -241,6 +245,26 @@ type RefreshToken struct {
 	Nonce string
 }
 
+// RefreshTokenRef is a reference object that contains metadata about refresh tokens.
+type RefreshTokenRef struct {
+	ID        string
+	CreatedAt time.Time
+	LastUsed  time.Time
```

Comment:
I put this as a part of this struct in case we want to list the refresh tokens per user and know the time-stamps related to each. In that case it will save us several database retrievals (per user refresh token).